main.py

```diff
@@ -1,173 +1,267 @@
+                        "1",
+                        "-",
+                    ],
+                    stdin=subprocess.PIPE,
+                    start_new_session=True,
+                )
+                play_process.communicate(input=audio)
+                play_process = None
+
         except Exception as e:
-            print("error", e)
-            notify("TTS-Reader: error playing text :(")
+            print(e)
+            notify("Failed to play")
+
         finally:
-            os.remove(os.path.join(audio_file_path, file_name))
+            pass
 
 
-readThread = threading.Thread(target=_process_read_text, daemon=True)
-readThread.start()
+play_thread = threading.Thread(target=thread_play, daemon=True)
+play_thread.start()
 
 
 @app.route("/read")
 def read():
-    add_text()
-    return ""
-
-
-def sanitizeText(text: str):
-    """
-    This function sanitizes the input text by removing non-ASCII characters and hyphenation artifacts. Example:
-    - 𝗟𝗮𝘁𝗲𝗻𝗰𝘆 -> Latency
-    - in pdf files " better perfor‐\nmance"
+    global stop_event
+    stop_event.clear()
 
-    Parameters:
-    text (str): The input string to be sanitized.
+    num_chars = 0
 
-    Returns:
-    str: The sanitized text.
-    """
-    text = unidecode(text)
-    text = text.replace("‐\n", "")
-    text = text.replace("‐ ", "")
-    return text
-
-
-def add_text():
-    global tokens
     try:
-        is_wayland = bool(parser.parse_args().wayland)
-        if is_wayland:
-            out_binary = subprocess.check_output(["wl-paste", "-p"])
-        else:
-            out_binary = subprocess.check_output(["xclip", "-o", "-selection primary"])
-        text: str = out_binary.decode("utf-8")
+        out = subprocess.check_output(
+            ["wl-paste", "-p"]
+            if parsed.wayland
+            else ["xclip", "-o", "-selection primary"]
+        )
+        text = out.decode("utf-8")
+        num_chars = len(text)
+
     except Exception as e:
         print(e)
-        notify("Unable to get selected text")
+        notify("Failed to get selected text")
         return
-    tokens = text.split(". ")
+
     try:
-        while tokens:
-            text = tokens[0].strip() + "."
-            tokens = tokens[1:]
-            text = sanitizeText(text)
-            file_name = f"{uuid.uuid4()}.wav"
-            process = Popen(
-                [
-                    "./script.sh",
-                    f"{file_name}",
-                    f'"{text}"',
-                    parser.parse_args().model,
-                    parser.parse_args().model_config,
-                ]
-            )
-            process.wait()
-            queue.append(file_name)
+        if not parsed.full_selection:
+            tokens = text.split(". ")
+            while tokens:
+                text = tokens[0].strip() + "."
+                tokens = tokens[1:]
+                text = sanitize_text(text)
+
+                out = generate_audio(text)
+                if out != None:
```

Comment:
```suggestion
                if out is not None:
```
This is more pythonic. If you have some time we can change it accordingly.

Reply:
Did not know about that. Changing to is in all locations